src/components/EditorsColumn.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import isEmpty from 'lodash/isEmpty';
+import includes from 'lodash/includes';
+import partial from 'lodash/partial';
+
+import EditorContainer from './EditorContainer';
+import Editor from './Editor';
+
+function allErrorsFor(language, errors, runtimeErrors) {
+  if (language === 'javascript') {
+    return errors.javascript.items.concat(runtimeErrors);
+  }
+  return errors[language].items;
+}
+
+export default function EditorsColumn({
+  currentProject,
+  errors,
+  onComponentMinimize,
+  onEditorInput,
+  onRequestedLineFocused,
+  runtimeErrors,
+  ui,
+}) {
+  const editors = [];
+  ['html', 'css', 'javascript'].forEach((language) => {
+    if (includes(ui.minimizedComponents, `editor.${language}`)) {
+      return;
+    }
+
+    editors.push(
+      <EditorContainer
+        key={language}
+        language={language}
+        source={currentProject.sources[language]}
+        onMinimize={
+          partial(onComponentMinimize, `editor.${language}`)
+        }
+      >
+        <Editor
+          errors={allErrorsFor(language, errors, runtimeErrors)}
+          key={language}
+          language={language}
+          percentageOfHeight={1 / editors.length}
+          projectKey={currentProject.projectKey}
+          requestedFocusedLine={ui.editors.requestedFocusedLine}
+          source={currentProject.sources[language]}
+          onInput={partial(onEditorInput, language)}
+          onRequestedLineFocused={onRequestedLineFocused}
+        />
+      </EditorContainer>,
+    );
+  });
+
+  if (isEmpty(editors)) {
+    return null;
+  }
+
+  return (
+    <div className="environment__column">
+      <div className="environment__columnContents editors">{editors}</div>
+    </div>
+  );
+}
+
+EditorsColumn.propTypes = {
+  currentProject: React.PropTypes.object.isRequired,
+  errors: React.PropTypes.object.isRequired,
+  runtimeErrors: React.PropTypes.array.isRequired,
+  ui: React.PropTypes.shape({
+    editors: React.PropTypes.object.isRequired,
+    minimizedComponents: React.PropTypes.array.isRequired,
+  }).isRequired,
```

Comment:
This is cool—I’m hoping to move to passing `Immutable` props all around rather than converting to JS objects on every state transition. Happily, [it looks like `react-immutable-proptypes` does support an equivalent to `shape`](https://github.com/HurricaneJames/react-immutable-proptypes), and also I’m very interested in making use of [`Immutable.Record`](https://facebook.github.io/immutable-js/docs/#/Record).